xbstrap/mirror/__init__.py

```diff
@@ -27,10 +29,11 @@ def do_update(args):
     if not args.no_update:
         plan.update = True
 
-    for src in cfg.all_sources():
-        plan.wanted.add((xbstrap.base.Action.MIRROR_SRC, src))
+    with xbstrap.util.lock_directory(path.join(cfg.build_root, "mirror")):
+        for src in cfg.all_sources():
+            plan.wanted.add((xbstrap.base.Action.MIRROR_SRC, src))
 
-    plan.run_plan()
+        plan.run_plan()
```

Comment:
I'd like to keep the locking outside the CLI handling.

It'd be better to take the lock within `mirror_src()` in base.py, i.e., when the "mirror" step is actually executed.

Reply:
wouldn't this break up the mirror lock between different packages in it? but as mentioned before, this might be a bigger issue, since the creation of the bootstrap.link file is also not caught in this lock, so maybe it'd be appropriate to also add a way to set the source directory other than readlink+dirname